R/bench-auroc.R

```diff
@@ -0,0 +1,132 @@
+#' This function takes the elements of the `activity` column and calculates
+#' precision-recall and ROC curves (depending on `curve`).
+#' The `activity` column is populated with the output for each stat method and
+#' results from the `run_benchmark()` function. Each of the elements
+#' in `activity` are the results from runs of the `decouple()` wrapper.
+#'
+#' @param df run_benchmark roc column provided as input
+#' @param downsampling logical flag indicating if the number of TN should be
+#'   downsampled to the number of TP
+#' @param times integer showing the number of downsampling
+#' @param ranked logical flag indicating if input is derived from composite
+#'   ranking that already took up-/downregulation (sign) into account
+#' @param curve whether to return a Precision-Recall Curve ("PR") or ROC ("ROC")
+#'
+#' @return tidy data frame with precision, recall, auc, n, tp, tn and coverage
+#'
+#' @import yardstick
+calc_curve = function(df, downsampling = T,
+                         times = 1000,
+                         ranked = F,
+                         curve="PR") {
+
+  if(curve=="PR"){
+    res_col_1 <- "precision"
+    res_col_2 <- "recall"
+    curve_fun = yardstick::pr_curve
+    auc_fun = yardstick::pr_auc
+  }
+  else if(curve=="ROC"){
+    res_col_1 <- "sensitivity"
+    res_col_2 <- "specificity"
+    curve_fun = yardstick::roc_curve
+    auc_fun = yardstick::roc_auc
+  }
+
+
+  if (ranked == T) {
+    df = df %>% prepare_for_roc(., filter_tn = T, ranked = T)
+  } else {
+    df = df %>%
+      prepare_for_roc(., filter_tn = T, ranked = F)
+  }
+
+  if (length(which(df$response == 0)) == nrow(df)){
```

Comment:
This can be replaced just with `sum(df$response == 0) == now(df)`
Also, this conditional has no unit test.